common/models/entities_with_tags.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+type EntitiesWithTags []*EntityWithTags
+
+type ewtFilter func(e EntityWithTags) bool
+
+func (e EntitiesWithTags) RemoveIf(f ewtFilter) EntitiesWithTags {
```

Comment:
Fair enough. My thought was more around if someone external was to consume this package individually, would the logic contained here be helpful? If it's purely specific to our implementation surrounding this package, I think we should move the logic into `common`.